lms/djangoapps/class_dashboard/tests/test_dashboard_data.py

```diff
@@ -85,6 +98,36 @@ def setUp(self):
                     state=json.dumps({'attempts': self.attempts}),
                 )
 
+            # adding an instructor and a staff to the site. They should not be counted. Test should not fail
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=instructor_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=staff_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            # Adding this student will fail the test
+
+            # StudentModuleFactory.create(
+            #     grade=1,
+            #     max_grade=1,
+            #     student=newstudent,
+            #     course_id=self.course.id,
+            #     module_state_key=self.item.location,
+            #     state=json.dumps({'attempts': self.attempts}),
+            # )
+
             for j, user in enumerate(self.users):
                 StudentModuleFactory.create(
```

Comment:
I found 3 more places caused by that sloppy bug where self.sub_section.location should be used instead of self.item.location.  These tests were all about 'sequentials'.  

I've fixed all of them and now all tests are PASSed.  

def test_get_students_opened_subsection(self):  
-    attributes = '?module_id=' + self.item.location.to_deprecated_string()
 
 
def test_get_students_opened_subsection_max(self):      
-    attributes = '?module_id=' + self.item.location.to_deprecated_string()


def test_get_students_opened_subsection_csv(self):  
-    attributes = '?module_id=' + self.item.location.to_deprecated_string() + '&tooltip=' + tooltip + '&csv=true'